### PR TITLE
DOC: add missing "regularized" to `gammainccinv` documentation

### PR DESCRIPTION
**Pull request by iamDecode**
#### Reference issue
None (I can make a separate issue if required)

#### What does this implement/fix?
The documentation of `gammainccinv` was incorrect: it mentioned it was the inverse of the upper gamma function without mentioning it was *regularized*.  This confused me a bit when looking for the correct method for one of my projects. This was introduced in #10690. Given `gammaincc` *is* regularized, and the following holds:

```python
gammainccinv(1, gammaincc(1, i)) == i
```

I conclude the documentation is incorrect and needs to be updated. In this PR I also fixed a related typo of the word "regularized" elsewhere (L4225).

Note the inverse of the *lower* gamma function (`gammaincinv`) does mentioned the regularization correctly:
https://github.com/scipy/scipy/blob/b6481610941b62d23bdd9a104840b58da35cac0b/scipy/special/_add_newdocs.py#L4218

#### Additional information
This is my first contribution so not fully familiar with the process, but I suspect this change does not affect test cases or documentation rendering in any way.

### CODE REVIEW

File: scipy/special/_add_newdocs.py [hunk 1]
```diff
@@ -4155,13 +4155,14 @@ def add_newdoc(name, doc):
     """
     gammainccinv(a, y)
 
-    Inverse of the upper incomplete gamma function with respect to `x`
+    Inverse of the regularized upper incomplete gamma function with
+    respect to `x`
```
Comment (rgommers) [inline scipy/special/_add_newdocs.py:4159]: How about we delete "with respect to x" here (and also in the `gammaincinv` docstring below? There's a couple of reasons: it's confusing because there is no input `x`, it doesn't add anything useful to what's written further down, and the single line is now too long which makes the summary table content overflow.

**Review by rgommers — COMMENTED** [review]
Thanks @iamDecode. This looks good, just one suggestion for a further cleanup.

File: scipy/special/_add_newdocs.py (hunk 1 above)
Comment (iamDecode) [reply scipy/special/_add_newdocs.py]: Hmm, it *does* seem that most descriptions of inverse methods in `scipy.special` do mention 'with respect to `<var>`", where `<var>` is a parameter of the method to be inversed. In case of `gammaincc(a, x)`, the original method has two parameters: the inverse may be with respect to either `a` or `x`. However, the "with respect to" part is definitely the least important part of the description here, and it is described later in the long description anyway.

so tldr; I agree, added in ed587fb097eeceb1cc8c804a659cc18347b28f22 :)

**Review by rgommers — APPROVED** [review]
LGTM, thanks @iamDecode!
